app/models/piece.rb

```diff
@@ -5,7 +5,52 @@ class Piece < ActiveRecord::Base
   validates :color, inclusion: { in: %w(black white) }
   validates :type, inclusion: { in: %w(Pawn Rook Bishop Knight King Queen) }
 
-end
+  private
+
+  #ALL MOVE CHECKS ASSUME WHITE PLAYER IS AT BOTTOM OF 2-D ARRAY/BOARD, AND BLACK PLAYER IS ON TOP.  NO IMPACT ON VIEW. -Chad
+
+  def x_distance(x) #calculates how many tiles piece has moved on x-axis
+    (x_position - x).abs
+  end
+
+  def y_distance(y) #calculates how many tiles piece has moved on y-axis
```

Comment:
I went back and forth on this one and I found that x_distance(x) and y_distance(y) were a bit easier to remember and write.  I just refactored it to distance(x, "x") and distance(y,"y") which is one function but takes in an additional parameter to look up the right home coordinate but I find it a bit more clunky to read.